libraries/networking/src/AddressManager.cpp

```diff
@@ -358,11 +358,9 @@ bool AddressManager::handleUrl(const QUrl& lookupUrlIn, LookupTrigger trigger) {
         emit lookupResultsFinished();
 
         return true;
-    } else if (lookupUrl.scheme() == HIFI_URL_SCHEME_FILE) {
-        // TODO -- once Octree::readFromURL no-longer takes over the main event-loop, serverless-domain urls can
-        // be loaded over http(s)
-        // lookupUrl.scheme() == URL_SCHEME_HTTP ||
-        // lookupUrl.scheme() == HIFI_URL_SCHEME_HTTPS ||
+    } else if (lookupUrl.scheme() == HIFI_URL_SCHEME_FILE || lookupUrl.scheme() == HIFI_URL_SCHEME_HTTP
+            || lookupUrl.scheme() == HIFI_URL_SCHEME_HTTPS) {
+        
```

Comment:
Did you verify that readFromURL no longer takes over the main-event loop? What happens if we have a file that is huge, or a page that times out?

Reply:
It still takes over the main event loop, hence the "do not merge", for now.
Am working on making the reading asynchronous, to remove the use of the event loop.